test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Services/PackageVulnerabilityServiceTests.cs

```diff
@@ -0,0 +1,262 @@
+            List<IReadOnlyDictionary<string, IReadOnlyList<PackageVulnerabilityInfo>>> knownVulnerabilities = new List<IReadOnlyDictionary<string, IReadOnlyList<PackageVulnerabilityInfo>>>()
+            {
+                new Dictionary<string, IReadOnlyList<PackageVulnerabilityInfo>>
+                {
+                    {
+                        "Package1",
+                        new PackageVulnerabilityInfo[] {
+                            new PackageVulnerabilityInfo(new Uri("https://vulnerability1"), PackageVulnerabilitySeverity.Low, VersionRange.Parse("[1.0.0,2.0.0)"))
+                        }
+                    }
+                }
+            };
+
+            GetVulnerabilityInfoResult getVulnerabilityInfoResult = new GetVulnerabilityInfoResult(knownVulnerabilities, exceptions: null);
+
+            int getVulnerabilityCallCount = 0;
+
+            var vulnerabilityResource = new Mock<IVulnerabilityInfoResource>();
+            vulnerabilityResource.Setup(resource => resource.GetVulnerabilityInfoAsync(
+                It.IsAny<SourceCacheContext>(),
+                It.IsAny<ILogger>(),
+                It.IsAny<CancellationToken>()))
+                .Callback(() => getVulnerabilityCallCount++)
+                .ReturnsAsync(getVulnerabilityInfoResult);
+
+            var sourceRepository = new Mock<SourceRepository>();
+
+            sourceRepository.Setup(s => s.GetResourceAsync<IVulnerabilityInfoResource>(CancellationToken.None))
+                .ReturnsAsync(vulnerabilityResource.Object);
+
+            PackageVulnerabilityService packageVulnerabilityService = new PackageVulnerabilityService(new List<SourceRepository> { sourceRepository.Object }, _testLogger);
+
+            // Act & Assert
+            List<PackageVulnerabilityMetadataContextInfo> vulnerabilities = await packageVulnerabilityService.GetVulnerabilityInfoAsync(new PackageIdentity("Package1", new NuGetVersion("1.0.0")), CancellationToken.None);
+            Assert.True(vulnerabilities.Count == 1);
+            Assert.Equal(1, getVulnerabilityCallCount);
+            vulnerabilities = await packageVulnerabilityService.GetVulnerabilityInfoAsync(new PackageIdentity("Package1", new NuGetVersion("1.0.0")), CancellationToken.None);
+            Assert.True(vulnerabilities.Count == 1);
+            Assert.Equal(1, getVulnerabilityCallCount);
+            vulnerabilities = await packageVulnerabilityService.GetVulnerabilityInfoAsync(new PackageIdentity("Package1", new NuGetVersion("1.0.0")), CancellationToken.None);
+            Assert.True(vulnerabilities.Count == 1);
+            Assert.Equal(1, getVulnerabilityCallCount);
+        }
+
+
+        [Fact]
+        public async Task GetVulnerabilityInfoAsync_ResetVulnerabilityData_ResetsVulnerabilityData()
+        {
+            // Arrange
+            PackageSource source = new PackageSource("https://contoso.com/vulnerability/v3/index.json");
```

Comment:
```suggestion
            PackageSource source = new PackageSource("https://contoso.test/vulnerability/v3/index.json");
```

contoso.com is a real domain, and if someone in the future changes this test, they might accidentally make it make real HTTP requests to it.

Using the `.test` TLD, since it's a reserved TLD that will never get domain registrations, we can be confident that CI won't spam requests to a real server, the test will fail with a DNS resolution error.